### PR TITLE
docs: add comments

### PR DESCRIPTION
**Pull request by bintangoding**
-Documentation Improvements
- This change was necessary to improve code documentation and enhance its clarity for developers and contributors

### CODE REVIEW

File: application/app.py [hunk 1]
```diff
@@ -1,43 +1,42 @@
+# import necessary modules
```
Comment (dartpain) [inline application/app.py:1]: This is probably unecessary

File: application/app.py [hunk 2]
```diff
@@ -1,43 +1,42 @@
...
 from application.api.internal.routes import internal
 
-
-
 # Redirect PosixPath to WindowsPath on Windows
-
 if platform.system() == "Windows":
     import pathlib
 
+    # Temporary alias to handle PosixPath on Windows
```
Comment (dartpain) [inline application/app.py:15]: Unecessary

File: application/app.py [hunk 3]
```diff
@@ -1,43 +1,42 @@
...
     temp = pathlib.PosixPath
     pathlib.PosixPath = pathlib.WindowsPath
 
-# loading the .env file
+# loading envirornment variables from a .env file
```
Comment (dartpain) [inline application/app.py:19]: Same

File: application/app.py [hunk 4]
```diff
@@ -1,43 +1,42 @@
...
 dotenv.load_dotenv()
 
-
-
+# Create a Flask application instance
```
Comment (dartpain) [inline application/app.py:22]: Unecessary

File: application/app.py [hunk 5]
```diff
@@ -51,10 +50,7 @@ def home():
         # Handle other cases or render the default page
```
Comment (dartpain) [inline application/app.py:50]: Unecessary

File: application/app.py [hunk 6]
```diff
@@ -63,6 +59,6 @@ def after_request(response):
     # response.headers.add("Access-Control-Allow-Credentials", "true")
     return response
 
-
+# Start the flask application if this script is executed directly
```
Comment (dartpain) [inline application/app.py:62]: Unecessary

**Review by dartpain — CHANGES_REQUESTED** [review]
Im very happy that you chose to contribute to docsgpt, I left a review with comments, I just think you need to comment things that add more value to a reader and not create unecessary burden of text. I suggest you check out application/api/answer/routes.py file I think it needs extra comments, this would be very appreciated.

**Comment by larinam** [comment]
From my perspective all changes in this PR are excessive. 

```A good best practice to follow when writing comments in code is to describe the why, not the what. In general, good code should be self-explanatory in how it works, at least in isolation. Explain the reasoning behind the code, not just what the code is doing. This can help others understand the intent of the code.```

https://swimm.io/learn/code-collaboration/comments-in-code-best-practices-and-mistakes-to-avoid

**Pull request closed without merging**